contracts/GenericLender/GenericAave.sol

```diff
@@ -129,47 +260,98 @@ contract GenericAave is GenericLenderBase {
         return looseBalance;
     }
 
-    function deposit() external override management {
-        uint256 balance = want.balanceOf(address(this));
-        _lendingPool().deposit(address(want), balance, address(this), 7);
-    }
-
-    function withdrawAll() external override management returns (bool) {
-        uint256 invested = _nav();
-        uint256 returned = _withdraw(invested);
-        return returned >= invested;
-    }
+    function _deposit(uint256 amount) internal {
+        ILendingPool lp = _lendingPool();
+        // NOTE: check if allowance is enough and acts accordingly
+        // allowance might not be enough if 
+        //     i) initial allowance has been used (should take years)
+        //     ii) lendingPool contract address has changed (Aave updated the contract address)
+        if(want.allowance(address(this), address(lp)) < amount){
+            IERC20(address(want)).safeApprove(address(lp), type(uint256).max);
+        }
 
-    function hasAssets() external view override returns (bool) {
-        return aToken.balanceOf(address(this)) > 0;
+        lp.deposit(address(want), amount, address(this), 179);
```

Comment:
should the referral code be harcoded or a setting that can be updated to avoid redeploying?

Reply:
created a setter to be able to update it